### PR TITLE
More fish modularization, adds fish debug items, the fishing hat can now vore, you can fish in bathwater

### PR DESCRIPTION
**Pull request by Gboster-0**
## About The Pull Request

- Moves the fishing hat and cloak to the modular fishing folder, along with XP for the cloak

- Adds in fishing hat vore
(you can click on a fishing hat with another fishing hat for the first one to become bigger and gain special examine text, up to 20)
at its maximum 20 hats it occupies a 4x4 space tile-wise (only as an item, not when you have it on)

- Makes the turfs in piscine mermaid's containment actual water turfs, allowing you to fish in them!
Everything has been playtested and adjusted accordingly to be similar-ish to the way it was before, mermaid still girlbosses you

- Adds in 3 new debug items:

- debug fishing rod: allows you to set any fishing speed on it by clicking on it, now you can easily check if your code is correct by fishing with 0.1 second speed

- debug fishing bag: simply has a lot of space, much more than what you will need for 99.9% of tests... or anything in that matter

- debug fishing turf spawner: allows you to quickly spawn a freshwater turf when you click on a tile

## Why It's Good For The Game

- its easier to now change the cloak and hat since they are in the fishing folder

- People wanted fishing in mermaid bathwater, so thy shall have it

- having 99 fishing hats and nothing to do with them was a bit boring, vore is the clear solution

- its now easier to also debug stuff with the new items, quite the positive for our shrimp devs

video for demonstration of why its good:

https://github.com/vlggms/lobotomy-corp13/assets/82319946/9de03992-41f9-4aa9-b92b-13d5dbbc6c72

## Changelog
:cl:
add: You can now feed your fishing hats to other fishing hats for them to become bigger!
add: You can now fish in mermaid's bathwater.
code: the fishing cloak and hat are now in the modular fishing folder
admin: added more things to the fishing crate, now you too can test fish in seconds!
/:cl:

### CODE REVIEW

File: ModularTegustation/fishing/code/fishing_items/fishing_hat.dm
Comment (LanceSmites328) [inline ModularTegustation/fishing/code/fishing_items/fishing_hat.dm:1]: The true potential of this hat.

File: ModularTegustation/fishing/code/turfs.dm [hunk 1]
```diff
@@ -63,6 +65,8 @@
 	. = ..()
 	if(!target_turf || is_type_in_typecache(thing, forbidden_types) || (thing.throwing && !istype(thing, /obj/item/food/fish || /obj/item/aquarium_prop )) || (thing.movement_type & (FLOATING|FLYING))) //replace this with a varient of chasm component sometime.
 		return
+	if(safe) // if this turf is in fact safe (currently only used for piscine mermaid)
+		return
```
Comment (LanceSmites328) [inline ModularTegustation/fishing/code/turfs.dm:69]: Move this above the type checks, but after the `. = ..()`. Putting it there reduces quantity of checks being made.

**Review by LanceSmites328 — CHANGES_REQUESTED** [review]

File: ModularTegustation/fishing/code/turfs.dm (hunk 1 above)
Comment (Gboster-0) [reply ModularTegustation/fishing/code/turfs.dm:69]: i'll assume ya mean
```
/turf/open/water/deep/Entered(atom/movable/thing, atom/oldLoc) //Sinking Code
	. = ..()
	if(safe) // if this turf is in fact safe (currently only used for piscine mermaid)
		return
	if(!target_turf || is_type_in_typecache(thing, forbidden_types) || (thing.throwing && !istype(thing, /obj/item/food/fish || /obj/item/aquarium_prop )) || (thing.movement_type & (FLOATING|FLYING))) //replace this with a varient of chasm component sometime.
		return
```
if so, done

File: code/modules/mob/living/simple_animal/abnormality/he/pisc_mermaid.dm [hunk 2]
```diff
@@ -193,23 +185,30 @@
...
+	var/time_strangled = 0
+	while(petter.stat != DEAD)
+		if(time_strangled > 2 MINUTES) // if they live for 2 minutes, make sure they are not alive and reset them
+			petter.losebreath += 500
+			petter.move_resist = MOVE_RESIST_DEFAULT
+			petter.pull_force = PULL_FORCE_DEFAULT
+			break
+		petter.adjustOxyLoss(3, updating_health=TRUE, forced=TRUE)
+		time_strangled++
+		SLEEP_CHECK_DEATH(1 SECONDS)
```
[the remark this answers is not in the document]
Comment (Gboster-0) [reply code/modules/mob/living/simple_animal/abnormality/he/pisc_mermaid.dm:211]: this probably works?
```
	while(petter.stat != DEAD)
		if(time_strangled > 2 MINUTES) // if they live for 2 minutes, make sure they are not alive and reset them
			petter.losebreath += 500
			petter.move_resist = MOVE_RESIST_DEFAULT
			petter.pull_force = PULL_FORCE_DEFAULT
			break
		if(petter.stat == DEAD)
			petter.move_resist = MOVE_RESIST_DEFAULT
			petter.pull_force = PULL_FORCE_DEFAULT
			break
		petter.adjustOxyLoss(3, updating_health=TRUE, forced=TRUE)
		time_strangled++
		SLEEP_CHECK_DEATH(1 SECONDS)
```

[the remark this answers is not in the document]
Comment (LanceSmites328) [reply code/modules/mob/living/simple_animal/abnormality/he/pisc_mermaid.dm]: Yea yea

**Review by Kitsunemitsu — APPROVED** [review]
Looks good to me.
Great PR.

**Review by LanceSmites328 — APPROVED** [review]

**Pull request merged**